src/ngraph/runtime/cpu/cpu_debugger.cpp

```diff
@@ -108,3 +132,60 @@ void* runtime::cpu::CPU_Debugger::inspect(std::shared_ptr<Node> op, size_t outpu
     return m_callframe.m_external_function->tensor_data.at(op->get_name() + "_" +
                                                            to_string(output_index));
 }
+
+bool runtime::cpu::CPU_Debugger::add_tracepoint(
+    std::shared_ptr<Node> op, const std::function<void(void**, const std::string&)>& callback)
+{
+    auto external_function = m_callframe.m_external_function;
+    bool found;
+    size_t pc;
+    std::tie(found, pc) = find_pc_for_node(op);
+    if (found)
+    {
+        if (replaced_functors.count(pc) != 0)
+        {
+            return false;
+        }
+
+        auto op_name = op->get_name();
+        std::vector<void**> poutputs;
+        for (size_t i = 0; i < op->get_outputs().size(); i++)
+        {
+            poutputs.push_back(&external_function->tensor_data.at(op_name + "_" + to_string(i)));
+        }
+
+        auto original_functor = external_function->functors.at(pc);
+
+        auto trace_functor =
+            [poutputs, callback, original_functor, op_name](CPURuntimeContext* ctx) {
+                original_functor(ctx);
+
+                std::vector<void*> outputs;
+                for (auto pout : poutputs)
+                {
+                    outputs.push_back(*pout);
+                }
+
+                callback(outputs.data(), op_name);
```

Comment:
would be even more awesome if we can reorder the data to row-major layouts (in a future PR).